woocommerce/woocommerce-android
# Get origin address network

Closes: #12437 

### Description
This PR adds the logic to connect the origin addresses request response with the UI. To achieve this goal, I:
- Created the address data store to cache origin addresses in a separate file and prevent retriggering requests when the store options get refreshed
- Refactored the store options data store so that the name reflects that the class is in charge of storing the shipping label configuration
- Created the new package `address` to group all the classes related to addresses
- Update the rest client and repository classes to include the request for fetching origin addresses
- Added the fetch origin addresses use case
- Added the observe origin addresses use case

### Testing information
TC1
1. Open the orders section
2. Open an order eligible for shipping label creation
3. Tap on create shipping label
4. Check that the origin addresses are fetched from the API
5. Navigate back
6. Tap again on create shipping label
7. Check that the origin addresses are presented from the preferences
8. Check that the origin addresses are refreshed on background

TC2
1. Open the orders section
2. Open an order eligible for shipping label creation
3. Tap on create shipping label
4. Check that the origin addresses are fetched from the API
5. Check that when the API request fails then the error screen is displayed

- You can apply the below path to force the fetch settings to fail

<details>
  <summary>Patch</summary>

```
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/address/FetchOriginAddresses.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/address/FetchOriginAddresses.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/address/FetchOriginAddresses.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/address/FetchOriginAddresses.kt	(revision 2783940e973d9c96722e7897e87c08da8e9654ae)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/address/FetchOriginAddresses.kt	(date 1735306410793)
@@ -10,20 +10,6 @@
     private val selectedSite: SelectedSite
 ) {
     suspend operator fun invoke(): Result<List<OriginShippingAddress>> {
-        return selectedSite.getOrNull()?.let {
-            val response = shippingRepository.fetchOriginAddresses(it)
-            val result = response.model
-            when {
-                response.isError.not() && !result.isNullOrEmpty() -> {
-                    Result.success(result)
-                }
-
-                else -> {
-                    val message =
-                        response.error?.message ?: if (result.isNullOrEmpty()) "Empty result" else "Unknown error"
-                    Result.failure(Exception(message))
-                }
-            }
-        } ?: Result.failure(Exception("No site selected"))
+        return  Result.failure(Exception("No site selected"))
     }
 }

```
</details>

### The tests that have been performed
- Checked that the UI works as expected when there is no cached data
- Checked that the UI works as expected when there is cached data
- Checked that the UI works as expected when refreshing the data fails

### Images/gif

https://github.com/user-attachments/assets/8f6ffacf-9827-46da-8103-ea5a4d3df2d3



## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
